### DISCUSSION TITLE
Compiling turns error unknown type name 'esp_lcd_panel_io_callbacks_t'

### DISCUSSION THREAD

**Discussion by AlexSchaeferBs** [rzeldent/esp32-smartdisplay#106] [Q&A]
When cloning (recursively) the demo project and trying to compile it (with PlatformIO) for the esp32-8048S050C produces an error `unknown type name 'esp_lcd_panel_io_callbacks_t'`

Can anyone help me out here? What am I missing?

Full compile log:

> `Processing esp32-8048S050C (board: esp32-8048S050C; platform: espressif32; framework: arduino)
> -----------------------------------------------------------------------------------------------------------------------------------------
> Verbose mode can be enabled via `-v, --verbose` option
> CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32-8048S050C.html
> PLATFORM: Espressif 32 (5.3.0) > esp32-8048S050C
> HARDWARE: ESP32S3 240MHz, 320KB RAM, 16MB Flash
> DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-builtin, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
> PACKAGES: 
>  - framework-arduinoespressif32 @ 3.20006.221224 (2.0.6) 
>  - tool-esptoolpy @ 1.40400.0 (4.4.0) 
>  - toolchain-riscv32-esp @ 8.4.0+2021r2-patch5 
>  - toolchain-xtensa-esp32s3 @ 8.4.0+2021r2-patch5
> LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
> LDF Modes: Finder ~ chain, Compatibility ~ soft
> Found 37 compatible libraries
> Scanning dependencies...
> Dependency Graph
> |-- esp32_smartdisplay @ 2.0.4+sha.bbe5984
> |-- ESP32-audioI2S @ 2.0.0+sha.e6ee006
> |-- lvgl @ 8.3.9
> Building in release mode
> Compiling .pio/build/esp32-8048S050C/lib5bb/esp32_smartdisplay/esp_lcd_panel_io_3wire_spi.c.o
> Compiling .pio/build/esp32-8048S050C/lib5bb/esp32_smartdisplay/esp_lcd_touch_gt911.c.o
> Compiling .pio/build/esp32-8048S050C/lib5bb/esp32_smartdisplay/esp_lcd_touch_xpt2046.c.o
> Compiling .pio/build/esp32-8048S050C/lib5bb/esp32_smartdisplay/lvgl_cst816s.c.o
> Compiling .pio/build/esp32-8048S050C/lib5bb/esp32_smartdisplay/lvgl_gc9a01.c.o
> Compiling .pio/build/esp32-8048S050C/lib5bb/esp32_smartdisplay/lvgl_gt911.c.o
> Compiling .pio/build/esp32-8048S050C/lib5bb/esp32_smartdisplay/lvgl_ili9341.c.o
> Compiling .pio/build/esp32-8048S050C/lib5bb/esp32_smartdisplay/lvgl_st7262.c.o
> Compiling .pio/build/esp32-8048S050C/lib5bb/esp32_smartdisplay/lvgl_st7701.c.o
> .pio/libdeps/esp32-8048S050C/esp32_smartdisplay/src/esp_lcd_panel_io_3wire_spi.c:69:88: error: unknown type name 'esp_lcd_panel_io_callbacks_t'
>  static esp_err_t panel_io_register_event_callbacks(esp_lcd_panel_io_handle_t io, const esp_lcd_panel_io_callbacks_t *cbs, void *user_ctx);
>                                                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
> .pio/libdeps/esp32-8048S050C/esp32_smartdisplay/src/esp_lcd_panel_io_3wire_spi.c: In function 'esp_lcd_new_panel_io_3wire_spi':
> .pio/libdeps/esp32-8048S050C/esp32_smartdisplay/src/esp_lcd_panel_io_3wire_spi.c:124:19: error: 'esp_lcd_panel_io_t' {aka 'struct esp_lcd_panel_io_t'} has no member named 'register_event_callbacks'
>      panel_io->base.register_event_callbacks = panel_io_register_event_callbacks;
>                    ^
> .pio/libdeps/esp32-8048S050C/esp32_smartdisplay/src/esp_lcd_panel_io_3wire_spi.c: At top level:
> .pio/libdeps/esp32-8048S050C/esp32_smartdisplay/src/esp_lcd_panel_io_3wire_spi.c:255:88: error: unknown type name 'esp_lcd_panel_io_callbacks_t'
>  static esp_err_t panel_io_register_event_callbacks(esp_lcd_panel_io_handle_t io, const esp_lcd_panel_io_callbacks_t *cbs, void *user_ctx)
>                                                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
> Compiling .pio/build/esp32-8048S050C/lib5bb/esp32_smartdisplay/lvgl_st7789.c.o
> Compiling .pio/build/esp32-8048S050C/lib5bb/esp32_smartdisplay/lvgl_st7796.c.o
> *** [.pio/build/esp32-8048S050C/lib5bb/esp32_smartdisplay/esp_lcd_panel_io_3wire_spi.c.o] Error 1`

**Comment by rzeldent** [comment]
Maybe try to update the platformio platforms?
This is "new" functionality in the IDF.

**Reply by AlexSchaeferBs** [reply]
PlatformIO is already up to date `6.1.13`. I suggest continuing under https://github.com/rzeldent/esp32-smartdisplay-demo/issues/14?!

**Reply by AlexSchaeferBs** [reply]
Any news or hints on how to solve this issue?

**Reply by rzeldent** [reply]
Hi Alex,

I suggest to use the precompiled files from the release. https://github.com/rzeldent/esp32-smartdisplay-demo/actions/runs/9828688543
Also, take a look at the automated build. Have you updated platformio? It is not just the version but also the esp32 libraries...
